dm/dm/worker/server.go

```diff
@@ -340,6 +340,10 @@ func (s *Server) observeRelayConfig(ctx context.Context, rev int64) error {
 					}
 					rev = rev1
 					if relaySource == nil {
+						if w := s.getWorker(false); w != nil && w.startedRelayBySourceCfg {
```

Comment:
`s` is not locked here. Should we use lock?
```suggestion
						if w := s.getWorker(true); w != nil && w.startedRelayBySourceCfg {
```